lib/jack_tokenizer.rb

```diff
@@ -9,8 +9,11 @@ def initialize(io)
   def has_more_tokens?
     loop do
       old_index = @index
-      skip_whitespace
-      skip_comments
+
+      if (match = @input.match(%r{[ \n]+|//.*$|/\*(.|\n)*\*/}, @index)) && match.begin(0) == @index
```

Comment:
Note that we had to replace `[ \n]*` with `[ \n]+` to make this single combined expression work; `[ \n]*` matches the empty string so it was otherwise preventing comment skipping from working in some cases.